api/client/client.go

```diff
@@ -22,12 +22,17 @@ type Client struct {
 }
 
 // NewClient starts the client.
-func NewClient(maddr ma.Multiaddr) (*Client, error) {
-	addr, err := util.TCPAddrFromMultiAddr(maddr)
-	if err != nil {
-		return nil, err
+func NewClient(target string, creds credentials.TransportCredentials) (*Client, error) {
+	var opts []grpc.DialOption
+	auth := tokenAuth{}
+	if creds != nil {
```

Comment:
If no creds are given, ie, during development / tests, just fallback to insecure.